talkfilters.man
.TH talkfilters 6 "1 May 2025" "Version 2.4"
.SH NAME
b1ff, brooklyn, chef, cockney, drawl, fudd, funetak, jethro, jive, kraut,
pansy, pirate, postmodern, redneck, valspeak, warez
\- GNU talk filters
.SH SYNOPSIS
\fIfilter\fP [\fB--version --help\fP]
.LP
.SH DESCRIPTION
.PP
The GNU Talkfilters are filter programs that convert ordinary English
text into text that mimics a stereotyped or otherwise humorous
dialect. Each program reads from standard input and writes to standard
output.

Since the filters do word and substring substitution on the text they
process, any word-wrap formatting of the original text will not be
preserved. Use the \fBwrap\fP filter to word-wrap the output of a talk 
filter.
.TP
.B austro
Austrian accent (Ahhhnold)
.TP
.B b1ff
B1FF of USENET yore
.TP
.B brooklyn
Brooklyn accent
.TP
.B chef
Swedish Chef (from The Muppet Show)
.TP
.B cockney
Londoner accent
.TP
.B drawl
Southern drawl
.TP
.B dubya
George "Dubya" Bush
.TP
.B fudd
Elmer Fudd (from the Looney Tunes cartoons)
.TP
.B funetak
Thick Asian accent
.TP
.B jethro
Jethro from The Beverly Hillbillies
.TP
.B jive
1970s Jive
.TP
.B kraut
German accent
.TP
.B pansy
Effeminate male
.TP
.B pirate
Pirate talk
.TP
.B postmodern
Postmodernist talk ("Feminazi")
.TP
.B redneck
Country redneck
.TP
.B valspeak
Valley talk
.TP
.B warez
H4x0r code
.SH SEE ALSO
.BR wrap (1),
.BR less (1)
.SH AUTHORS
Edward Betts, Clement Cole, John Hagerman, Brand Hilton, Adam Hudd,
Ian Johnston, Daniel V. Klein, Mark Lindner, Stephen K. Mulrine, Duane
Paulson, Samuel Stoddard, Matt Welsh, David Whitten, et al.


ChangeLog
2017-10-27  Mark Lindner  <markl@avalon>

	* *.l - fixed comment which caused flex error
	* <multiple> - cleaned up and updated build files

2007-12-29  Mark Lindner  <markl@neuromancer>

	* talkfilters.c, talkfilters.h, Makefile.am, Makefile.in, config.h,
	  - MinGW & VC++ build fixes
	* config.guess, config.sub, ltmain.sh, aclocal.m4, libtool
	  - updated to newest

2006-11-21  Mark Lindner  <markl@neuromancer>

	* pirate.l - minor improvements
	* dubya.l - minor improvements
	* common.h - use VERSION instead of hardcoded version number

2006-10-27  Mark Lindner  <markl@neuromancer>

	* common.c - fix compiler warning
	* *.sln, *.vcproj, win32/unistd.h - add Visual C++ Express 2005
	  build files
	* talkfilters.c - update version number
	* common.h - use C99 varargs macro notation; added some #defines for
	  win32
	* pirate.l - added rule
	* talkfilters.h - added extern "C" guard

2006-09-18  Mark Lindner  <markl@neuromancer>

	* common.c - fix compiler warnings
	* talkfilters.h, talkfilters.c, configure.in, Makefile.am
	  - changes for DLL creation

2006-04-17  Mark Lindner  <markl@neuromancer>

	* funetak.l - replaced random() with gtf_random(); eliminates build
	  error in mingw

2006-04-15  Mark Lindner  <markl@neuromancer>

	* <multiple> - added support for pkg-config

2005-11-21  Mark Lindner  <markl@neuromancer>

	* gaim_plugin/talkfilters.c, Makefile.am - Removed the GAIM plugin,
	  as it's now being maintained elsewhere.
	* common.c, talkfilters.c, common.h, configure.in - Build fixes.
	* talkfilters.spec - added new file
	The above all contributed by Peter Lawler

2005-01-10  Mark Lindner  <markl@neuromancer>

	* gaim_plugin/talkfilters.c - Updated to work with GAIM 1.1.0.

2004-06-28  Mark Lindner  <markl@neuromancer>

	* gaim_plugin/talkfilters.c - Incorporated bug fix submitted by a user.
	  The bug caused the IM to only be filtered locally (not sent in
	  filtered form to the other end). This must be a regression when
	  moving to GAIM 0.78/0.79.

2004-06-26  Mark Lindner  <markl@neuromancer>

	* gaim_plugin/* - Updated plugin to work with GAIM 0.79, and clarified
	  installation instructions.

2004-01-06  Mark Lindner  <markl@neuromancer>

	* funetak.l - Fixed a capitalization problem.
	* austro.l - Added missing rules, whose absence caused garbled output
	  from filter function.
	* gaim_plugin/* - New plugin for GAIM.

2003-12-02  Mark Lindner  <markl@neuromancer>

	* *.l, *.c, *.h - Added contributed austro and dubya filters.

2003-07-22  Mark Lindner  <markl@neuromancer>

	* *.l - Added rules to leave HTML tags intact. Do you see where
          I'm going with this? ;-)
	* kraut.l - Minor fixes and corrections (submitted by Thomas M. Ott)
	* talkfilters.man - Documentation updates.
	* configure.in - Rewrote `flex' check.

2003-04-29  Mark Lindner  <markl@neuromancer>

	* jive.l - Added a few more rules.

2003-02-13  Mark Lindner  <markl@dystance.net>

	* *.l - Made a fix so that compilation doesn't fail with older
	versions of gcc: moved declaration of _yybuf so that it's before the
	first statement in the filter function.

2003-02-11  Mark Lindner  <markl@dystance.net>

	* <multiple> - Complete overhaul of package; added automake/libtool
	support, and produce not only executables but also a library.

Tue Jul  4 21:33:08 2000  Mark A. Lindner <markl@dystance.net>

	* Makefile.in: Modified `dist' target and added `info' target.

	* talkfilters.texi: Texinfo documentation added.

Thu Dec 31 17:09:45 1998  Joel N. Weber II  <devnull@melange.gnu.org>

	* configure.in: Rewrote the check for lex to just check for flex
	and bail if it isn't found.

	* src/talkfilters-1.0/Makefile.in (LIBS): Hardcode -lfl, since we
	always use flex.



talkfilters.c
/* GNU Talkfilters
   Copyright (C) 1998-2025 Free Software Foundation, Inc.

   This file is part of GNU Talkfilters

   GNU Talkfilters is free software; you can redistribute it and/or
   modify it under the terms of the GNU General Public License as
   published by the Free Software Foundation; either version 2, or (at
   your option) any later version.

   This software is distributed in the hope that it will be amusing, but
   WITHOUT ANY WARRANTY; without even the implied warranty of
   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
   General Public License for more details.

   You should have received a copy of the GNU Library General Public
   License along with this library; if not, see
   <http://www.gnu.org/licenses/>.
*/

#ifdef HAVE_CONFIG_H
#include "config.h"
#endif

#include <stddef.h>
#ifdef HAVE_STRING_H
#include <string.h>
#endif

#include "talkfilters.h"

#ifdef WIN32
#define WIN32_LEAN_AND_MEAN
#include <windows.h>
#endif

#ifdef _MSC_VER
#pragma warning (disable: 4996)
#define strcasecmp stricmp
#endif

static const char *__gtf_version = "GNU Talkfilters v2.3.7";

extern int __gtf_filter_austro(const char *, char *, size_t);
extern int __gtf_filter_b1ff(const char *, char *, size_t);
extern int __gtf_filter_brooklyn(const char *, char *, size_t);
extern int __gtf_filter_chef(const char *, char *, size_t);
extern int __gtf_filter_cockney(const char *, char *, size_t);
extern int __gtf_filter_drawl(const char *, char *, size_t);
extern int __gtf_filter_dubya(const char *, char *, size_t);
extern int __gtf_filter_fudd(const char *, char *, size_t);
extern int __gtf_filter_funetak(const char *, char *, size_t);
extern int __gtf_filter_jethro(const char *, char *, size_t);
extern int __gtf_filter_jive(const char *, char *, size_t);
extern int __gtf_filter_kraut(const char *, char *, size_t);
extern int __gtf_filter_pansy(const char *, char *, size_t);
extern int __gtf_filter_pirate(const char *, char *, size_t);
extern int __gtf_filter_postmodern(const char *, char *, size_t);
extern int __gtf_filter_redneck(const char *, char *, size_t);
extern int __gtf_filter_valspeak(const char *, char *, size_t);
extern int __gtf_filter_warez(const char *, char *, size_t);

static gtf_filter_t __gtf_filters[] =
{
  { "austro",     "Austro",         __gtf_filter_austro     },
  { "b1ff",       "B1FF",           __gtf_filter_b1ff       },
  { "brooklyn",   "Brooklyn",       __gtf_filter_brooklyn   },
  { "chef",       "Swedish Chef",   __gtf_filter_chef       },
  { "cockney",    "London Cockney", __gtf_filter_cockney    },
  { "drawl",      "Southern Drawl", __gtf_filter_drawl      },
  { "dubya",      "George Dubya",   __gtf_filter_dubya      },
  { "fudd",       "Elmer Fudd",     __gtf_filter_fudd       },
  { "funetak",    "Funetak",        __gtf_filter_funetak    },
  { "jethro",     "Jethro",         __gtf_filter_jethro     },
  { "jive",       "Jive",           __gtf_filter_jive       },
  { "kraut",      "Kraut",          __gtf_filter_kraut      },
  { "pansy",      "Pansy",          __gtf_filter_pansy      },
  { "pirate",     "Pirate",         __gtf_filter_pirate     },
  { "postmodern", "Postmodernist",  __gtf_filter_postmodern },
  { "redneck",    "Redneck",        __gtf_filter_redneck    },
  { "valspeak",   "Valley Speak",   __gtf_filter_valspeak   },
  { "warez",      "Warez (H4x0r)",  __gtf_filter_warez      },
};

static int __gtf_filter_count = sizeof(__gtf_filters) / sizeof(gtf_filter_t);

/*
 */

#ifdef WIN32

BOOL WINAPI DllMain(HINSTANCE hinstDLL, DWORD fdwReason, LPVOID lpvReserved)
{
  return(TRUE);
}

#endif

/*
 */

int gtf_filter_count(void)
  {
  return(__gtf_filter_count);
  }

/*
 */

const gtf_filter_t *gtf_filter_list(void)
  {
  return(__gtf_filters);
  }

/*
 */

const gtf_filter_t *gtf_filter_lookup(const char *name)
  {
  int i;
  gtf_filter_t *f;

  for(i = 0, f = __gtf_filters; i < __gtf_filter_count; i++, f++)
    {
    if(!strcasecmp(f->name, name))
      return(f);
    }

  return(NULL);
  }

/* end of source file */
